doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+    material properties.
+
+Adding a new material
+---------------------
+Let's just add a subclass of sodium in our plugin that has a heat capacity defined. Make
+your new material in a new module called :file:`myapp/materials.py`:
+
+.. code-block:: python
+    :caption: ``myapp/materials.py``
+
+    from armi import materials
+    from armi.utils.units import getTc
+
+    class Sodium(materials.Sodium):
+        def heatCapacity(self, Tk=None, Tc=None):
+            """Sodium heat capacity in J/kg-K"""
+            Tc = getTc(Tc,Tk)
+            # not even temperature dependent for now
+            return 1.252
+
+But wait! Now there are **two** materials with the name *Sodium* in ARMI. Which will be
+chosen? ARMI uses a namespace order controlled by
+:py:func:`armi.materials.setMaterialNamespaceOrder` which can be set either
+programmatically (in an app) or at runtime (via the ``materialNamespaceOrder`` user
+setting). In our case, we want to set it at the app level, so we will add the following to
+the :file:`myapp/__init__.py` file:
+
+.. code-block:: python
+    :caption: Addition to ``_myapp/__init__.py``
+
+    from armi import materials
+    materials.setMaterialNamespaceOrder(
+        ["myapp.materials", "armi.materials"]
+    )
+
+Now ARMI should find our new updated Sodium material and get past that error.  Run it once
+again::
+
+    (armi) $ python -m myapp run anl-afci-177.yaml
+
+.. tip:: You may want to pipe the output to a log file for convenient viewing with
+    a command like ``python -m myapp run anl-afci-177.yaml > run.stdout``
+
+Checking the output
+===================
+Several output files should have been created in the run directory from that past command.
+Most important is the ``anl-afci-177.h5`` HDF5 binary database file. You can use this file
+to bring the ARMI state back to any state point from the run for analysis. 
+
+.. admonition:: Is there a general DB viewer?
+    TerraPower uses an internal HDF5 viewer called *XTVIEW* to view the state in the HDF5
+    database. At some point this tool will either be made available, or we or someone else
+    will create a plugin for a more generic visulaization tools like VisIT or Paraview.
+    For now you are stuck exploring the HDF5 output via the ARMI API. 
+
+A generic description of the outputs is provided in :doc:`/user/outputs/index`. 
+
+Adding new output
+-----------------
+Besides adding new state variables to the ARMI database, your plugins and applications may
+also find it beneficial to generate new output files, such as plots or other reports.
+Let's add automatic plots of coolant temperature at runtime in our application. 
+
+We can use the :ref:`detail assembly feature <detail-assems>` to allow users to
+choose which assembly will get printed out, since you may not want plots for every
+assembly.
+
+Open up the ``thermalSolver.py`` plugin code and add the following function to plot a
+single assembly:
```

Comment:
Alternative, there is an opportunity here to demonstrate the ease of use that the database interface offers for loading the reactor state and asking for the parameters based on the object at the locations of interest. This may be a nice way to demonstrate some more versatility in post-processing versus changing the underlying physics interface code.

Reply:
that's in another tutorial I believe.

Reply:
Another tutorial, and likely one that the user would have seen before this one.